airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -280,6 +277,149 @@ def test_secret_is_properly_marked(self, connector_spec_dict: dict, detailed_log
                     {properties}"""
             )
 
+    def test_property_type_is_not_array(self, connector_spec_dict: dict):
+        """
+        Each field has one or multiple types, but the UI only supports a single type and optionally "null" as a second type.
+        """
+        specification = connector_spec_dict["connectionSpecification"]
+        errors = []
+        for type_path, value in dpath.util.search(specification, "**/properties/*/type", yielded=True):
+            if isinstance(value, List):
+                if len(value) != 2:
```

Comment:
I think it would work but I would like to be strict here as there isn't a good reason type would just be a single-value list. One case less to worry about.

Reply:
I would prefer to still allow single value list. I agree there's no good reason to use it. 
But this might cause connector build failures on existing connectors that our teams will have to fix.
If it works with a single value list, let's not make the acceptance tests fail.

Reply:
As mentioned below I tested all existing sources, but if you want to be safe I'm happy to pull it in, no strong objections.